### ISSUE TITLE
Jobs launched with API error out (and stays in Pending state)

### ISSUE THREAD

**Issue by u1735067** [nautobot/nautobot#732]
### Environment
* Python version: 3.9
* Nautobot version: 1.1.0

### Steps to Reproduce
1. Login, create a job
2. Start a job using the API (example on the demo instance: `curl -X POST "https://demo.nautobot.com/api/extras/jobs/local%2Fdevice_validation%2FDeviceConnectionsReport/run/" -H  "accept: application/json" -H  "Content-Type: application/json" -H  "X-CSRFToken: fAHKXI87BV1KvBwS2DWjien2e2MiZIqOa8oP4fBG3ntyPJHS06uWUcNgwlBzi6pc" -d "{}"`

### Expected Behavior
The job is executed as in the webui

### Observed Behavior
Job error out (and stays blocked in Pending state, see https://demo.nautobot.com/extras/job-results/, but I filled another bug for this part)

### Additional informations

Observed in the logs on another instance:
```
celery_worker_1  | [2021-07-23 18:54:10,983: INFO/MainProcess] Task nautobot.extras.jobs.run_job[e43f026b-1361-4879-9906-19593fcc10f6] received
celery_worker_1  | 18:54:11.002 INFO    nautobot.jobs :
celery_worker_1  |   Repository successfully refreshed
celery_worker_1  | [2021-07-23 18:54:11,002: INFO/ForkPoolWorker-3] Repository successfully refreshed
celery_worker_1  | [2021-07-23 18:54:11,016: ERROR/ForkPoolWorker-3] Task nautobot.extras.jobs.run_job[e43f026b-1361-4879-9906-19593fcc10f6] raised unexpected: AttributeError("'str' object has no attribute 'items'")
celery_worker_1  | Traceback (most recent call last):
celery_worker_1  |   File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 450, in trace_task
celery_worker_1  |     R = retval = fun(*args, **kwargs)
celery_worker_1  |   File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 731, in __protected_call__
celery_worker_1  |     return self.run(*args, **kwargs)
celery_worker_1  |   File "/usr/local/lib/python3.9/site-packages/nautobot/extras/jobs.py", line 961, in run_job
celery_worker_1  |     data = job_class.deserialize_data(data)
celery_worker_1  |   File "/usr/local/lib/python3.9/site-packages/nautobot/extras/jobs.py", line 345, in deserialize_data
celery_worker_1  |     for field_name, value in data.items():
celery_worker_1  | AttributeError: 'str' object has no attribute 'items'
```

**Comment by u1735067** [comment]
So, the workaround: pass `{"data": {}}` in the body.

Cause: 
* https://github.com/nautobot/nautobot/blob/v1.1.0/nautobot/extras/views.py#L710 : using the UI, `data` is cleaned and a dict
* https://github.com/nautobot/nautobot/blob/v1.1.0/nautobot/extras/api/views.py#L274 / https://github.com/nautobot/nautobot/blob/v1.1.0/nautobot/extras/api/serializers.py#L510 : using the API, `data` is set to be a string (wrong) and defaults to an empty string.

**Comment by jathanism** [comment]
Thanks @u1735067 ... Looks like we need to fix the serializer for the payload.

**Comment by jmcgill298** [comment]
@u1735067 is this different than #651

**Comment by u1735067** [comment]
> @u1735067 is this different than #651

Thanks for pointing that out (I missed it but noticed the same as you later, when adding an optional field that was declared but not even present when launched from API), it's very similar yes, the only difference here is that even without fields in the job, if you completely omit the `data` field the call will crash. Maybe the difference is not worth keeping this issue but that's not up to me now :)

**Comment by jmcgill298** [comment]
okay, that is different. I'm not sure that is a bug, or expected behavior. Thanks for clarifying.

**Comment by glennmatthews** [comment]
Need to verify whether this is resolved by #828 or whether this is still an issue.

**Comment by romanukes** [comment]
Can't reproduce with "Device Validation" job, will run more tests and update

**Comment by jeffkala** [comment]
For documentation purposes.  I was doing an identical task on 1.1.2 (hitting this issue) and the identical task on 1.1.4 and I wasn't hitting this on 1.1.4.

**Comment by glennmatthews** [comment]
I think we can conclude that this was fixed by #828. Thanks @romanukes and @jeffkala for the data gathering!
